### PR TITLE
Feature/92 download files

### PR DESCRIPTION
**Pull request by FelixoGudiel**
This feature aims to fix two major problems in the app.

The first of them is the correct pathing for files. This feature can be checked by clicking in the link in the file field of an object. It should now trigger the download of the file (please upload a file first, as the populate does not contain the files referenced in the mock data (i can add my mock "files" folder so that it works, but i would have to change the gitignore for that)). This can be used for checking the submitted files or for the frontend showing of specific organizational files.

The second of them is the addition of the PATCH method to all api models. PATCH is a very interesting request, because it behaves in a similar way to PUT but with a quirk. A PUT request requires all fields in a `json` to update an object, as it completely overwrites the object. A PATCH, however, will only overwrite the given parameters in the `json`. This way, frontend can update file-related objects without having to upload all the files if they are unchanged. A new option PATCH shoud appear next to the PUT button in the api (given an id, example: meeting/2).

Please, check these functionalities and give me your insights on this theme!

### CODE REVIEW

File: nexong/views.py [hunk 1]
```diff
@@ -1 +1,19 @@
+from django.http import HttpResponse, Http404
+from django.conf import settings
+import os
+from mimetypes import guess_type
 
+
+def serve_file(request, path):
+    print(path)
```
Comment (auroranavas) [inline nexong/views.py:8]: Please, remove this print statement.

**Review by auroranavas — CHANGES_REQUESTED** [review]
Right now I am trying to make a PATCH to a CenterExit object but I am getting this error:
image
image

**Comment by FelixoGudiel** [comment]
Okay, the problem were the validators colliding with incomplete jsons. This has been fixed by checking the validators only if the field to validate is given in the json. Also I have eliminated PATCH permissions for every model that does not have a fileField, as it is problematic for the validators and this functionality is not essential in non fileField-classes. Thank you for your attention!

**Review by auroranavas — CHANGES_REQUESTED** [review]
Congrats on the PATH fix, it works wonderfully now. There is just a small problem with the files routing. Here's a screenshot. 

image

**Comment by FelixoGudiel** [comment]
You are right, when the two new classes (punctual donation and homedocument) were created, they took reference in the previous routes, that were not correct. This is the extract with the new classes with the incorrect fields. Now the FileField redirects to `proof_of_payment` and `home_document`.
image
Thank you! If you find anything else, tell me ASAP!

**Review by auroranavas — APPROVED** [review]
This is great, you did a really good job. Congrats and thank you for taking into account my feedback!

**Review by manortbla — APPROVED** [review]
Everything looks fine. Congratulations.

**Review by pabpercab1 — APPROVED** [review]
Functionality seems to work well. Well done!

**Pull request merged**